libs/remote_storage/src/s3_bucket.rs

```diff
@@ -532,7 +532,14 @@ impl RemoteStorage for S3Bucket {
                     .req_seconds
                     .observe_elapsed(kind, &response, started_at);
 
-                let response = response?;
+                let response = match response {
```

Comment:
on line L510R510 the continuation token is not cloned. As the setter seems to be by-value I do not understand how does this compile. Similar issue with azure I think. Maybe I am just slow.

Reply:
good catch, it indeed failed to compile after I applied your suggestion for `continue`. now I'm also cloning it.